src/Tribe/Commerce/PayPal/Frontend/Tickets_Form.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * Class Tribe__Tickets__Commerce__PayPal__Frontend__Tickets_Form
+ *
+ * @since TBD
+ */
+class Tribe__Tickets__Commerce__PayPal__Frontend__Tickets_Form {
+
+	/**
+	 * Whether the form has rendered already or not
+	 *
+	 * @var bool
+	 */
+	protected $has_rendered = false;
+
+	/**
+	 * @var Tribe__Tickets__Commerce__PayPal__Main
+	 */
+	protected $main;
+
+	/**
+	 * Tribe__Tickets__Commerce__PayPal__Frontend__Tickets_Form constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Tickets__Commerce__PayPal__Main $main
+	 */
+	public function __construct( Tribe__Tickets__Commerce__PayPal__Main $main ) {
+		$this->main = $main;
+	}
+
+	/**
+	 * Modifies the passed content to inject the front-end tickets form.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $content The post content
+	 */
+	public function render( $content ) {
+		if ( $this->has_rendered || ! $this->main->is_active() ) {
+			return $content;
```

Comment:
Knowing that this is merely lifted from another method and was perhaps done by me as a copypasta from elsewhere...

We are returning `$content` here but we don't return anything if we get to the end of the method - we are merely including a template. Should we instead capture the output and return it there? Or should this output `$content` here and then return? Or is the two different approaches the right way to go?

Just flagging it as slightly odd feeling. Since you are merely relocating this logic, don't feel like you need to fix/change it as part of this PR.